File: src/main/java/com/example/wizytydomowe/Patient/Patient.java
package com.example.wizytydomowe.Patient;

import com.example.wizytydomowe.Address.Address;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;
import org.hibernate.annotations.Cascade;
import org.hibernate.validator.constraints.pl.PESEL;

import javax.persistence.*;
import javax.validation.constraints.*;
import java.time.LocalDate;

@Entity
@Data
@Table(name = "patient")
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    @Size(min = 2)
    private String name;
    @NotNull
    @Size(min = 2)
    private String surname;
    @PESEL
    private String pesel;
    @Past
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate dateOfBirth;
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id")
    private Address address;
    @Digits(integer = 9, fraction = 0)
    private String phoneNumber;
    @Email
    private String email;
}

File: client/src/app/patient.ts
import {Address} from "./address";

export class Patient {
  name: string = '';
  surname: string = '';
  phoneNumber: string = '';
  pesel: string = '';
  email: string = '';
  address: Address;
  dateOfBirth: Date;


  constructor(name: string, surname: string, phoneNumber: string, pesel: string,
              email: string, address: Address, dateOfBirth: Date) {
    this.name = name;
    this.surname = surname;
    this.phoneNumber = phoneNumber;
    this.pesel = pesel;
    this.email = email;
    this.address = address;
    this.dateOfBirth = dateOfBirth;
  }
}
